wallet/src/actors/worker/methods.rs

```diff
@@ -135,6 +137,58 @@ impl Worker {
                 (id, default_account, Some(master_key))
             }
         };
+        // To know if the chain is valid, we have to wait for 10 blocks to build the chain and 10
+        // more blocks for its voting period. We could need fewer blocks if the current block is
+        // not the first of a superblock, so 20 is the safest value
+        let confirm_superblock_period: i64 =
+            (2 * self.params.consensus_constants.superblock_period + 1).into();
+
+        let birth_date = match birth_date {
+            Some(types::BirthDate::Current) => {
+                // Use the current epoch as birth date
+                let gen_fut = self.get_block_chain(0, -confirm_superblock_period);
+                let gen_res: Vec<ChainEntry> = futures::executor::block_on(gen_fut)?;
+                let gen_entry = gen_res
+                    .get(0)
+                    .expect("It should always found a superconsolidated block");
+                let get_gen_future = self.get_block(gen_entry.1.clone());
+                let (block, _confirmed) = futures::executor::block_on(get_gen_future)?;
+
+                CheckpointBeacon {
+                    checkpoint: block.block_header.beacon.checkpoint,
+                    hash_prev_block: block.hash(),
+                }
+            }
+            // Use provided epoch as birth date
+            Some(types::BirthDate::Imported(epoch)) => {
+                validate_birth_date(
+                    epoch,
+                    self.params.epoch_constants.checkpoint_zero_timestamp,
+                    self.params.epoch_constants.checkpoints_period,
+                )?;
+
+                let gen_fut = self.get_block_chain((epoch as i64) - confirm_superblock_period, 2);
```

Comment:
This doesn't check for overflow, `epoch - confirm_superblock_period` must never be negative

 ```suggestion
                let gen_fut = self.get_block_chain(epoch.saturating_sub(confirm_superblock_period), 2);
```